src/keystore.cpp

```diff
@@ -29,7 +29,12 @@ void CBasicKeyStore::ImplicitlyLearnRelatedKeyScripts(const CPubKey& pubkey)
 #ifdef DEBUG
         script = GetScriptForDestination(WitnessV0KeyHash(key_id));
 #else
-        script = GetScriptForDestination(key_id);
+        if (isColored) {
+            ColorIdentifier colorID = ColorIdentifier();
+            script = GetScriptForDestination(key_id, &colorID);
```

Comment:
In this case, the type of `colorID` is `TokenTypes::NONE`?

Reply:
because we will not know the colorId during add key so set to a default colorId